commands/files/readerfile.go

```diff
@@ -13,10 +13,12 @@ type ReaderFile struct {
 	fullpath string
 	reader   io.ReadCloser
 	stat     os.FileInfo
+	offset   int64
+	baseInfo ExtraInfo
 }
 
-func NewReaderFile(filename, path string, reader io.ReadCloser, stat os.FileInfo) *ReaderFile {
-	return &ReaderFile{filename, path, reader, stat}
+func NewReaderFile(filename, path, abspath string, reader io.ReadCloser, stat os.FileInfo) *ReaderFile {
```

Comment:
To be honest, i'm not entirely sure either. But that said, our tests in that area are quite thorough, If you change something, go into `test/sharness`, run `make deps` and then run `./t0040-add.sh`, and maybe some of the other `t004*` tests too to make sure nothing weird happened